### PR TITLE
[PM-5499] Create Auth Request Service

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Creates a new Auth Request Service to consolidate logic from Login Strategy Service and Auth Request Crypto Service. Removes Auth Request Crypto Service.

The only new logic in this PR is the tests for the new `AuthRequestService.approveOrDenyAuthRequest()`. The rest is copied and registration for the new service.
## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/desktop/src/auth/login/login-approval.component.ts
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/auth/login/login-approval.component.ts:1]: 🏅 , The rename of  `passwordlessLogin` to `approveOrDenyAuthRequest` brings great clarity. Nice work!

File: apps/desktop/src/auth/login/login-approval.component.ts [hunk 1]
```diff
@@ -121,10 +121,9 @@ export class LoginApprovalComponent implements OnInit, OnDestroy {
         this.i18nService.t("thisRequestIsNoLongerValid"),
       );
     } else {
-      const loginResponse = await this.loginStrategyService.passwordlessLogin(
-        this.authRequestResponse.id,
-        this.authRequestResponse.publicKey,
+      const loginResponse = await this.authRequestService.approveOrDenyAuthRequest(
```
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/auth/login/login-approval.component.ts:124]: Nit: could update `loginResponse` name as well as the call to update an auth req isn't a login response.

File: libs/auth/src/common/abstractions/auth-request.service.abstraction.ts
Comment (JaredSnider-Bitwarden) [inline libs/auth/src/common/abstractions/auth-request.service.abstraction.ts:1]: Nice work on these!

File: libs/auth/src/common/services/auth-request/auth-request.service.spec.ts [hunk 2]
```diff
@@ -1,32 +1,71 @@
...
-    authReqCryptoService = new AuthRequestCryptoServiceImplementation(cryptoService);
+    sut = new AuthRequestService(appIdService, cryptoService, apiService, stateService);
 
     mockPrivateKey = new Uint8Array(64);
   });
 
-  it("instantiates", () => {
-    expect(authReqCryptoService).not.toBeFalsy();
-  });
+  describe("approveOrDenyAuthRequest", () => {
```
Comment (JaredSnider-Bitwarden) [inline libs/auth/src/common/services/auth-request/auth-request.service.spec.ts:33]: Nit: should we have a happy path test?

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Nice work!

File: apps/desktop/src/auth/login/login-approval.component.ts (hunk 1 above)
Comment (jlf0dev) [reply apps/desktop/src/auth/login/login-approval.component.ts:124]: Will do in followup PR!

File: libs/auth/src/common/services/auth-request/auth-request.service.spec.ts (hunk 2 above)
Comment (jlf0dev) [reply libs/auth/src/common/services/auth-request/auth-request.service.spec.ts:33]: Will do in followup PR!

**Pull request merged**